File: services/data-pipeline/scripts/check_db.py
import os
import asyncio
from supabase import create_client, Client
from dotenv import load_dotenv

annotator_env_path = os.path.abspath(os.path.join(
    os.path.dirname(__file__), 
    "..", "..", "..", "apps", "expert-annotator", ".env"
))
load_dotenv(annotator_env_path)

url: str = os.environ.get("VITE_SUPABASE_URL")
key: str = os.environ.get("VITE_SUPABASE_ANON_KEY")

supabase: Client = create_client(url, key)

res = supabase.table("papers").select("id, title, filename").execute()
print("Papers in DB:")
for r in res.data:
    print(r)

File: services/data-pipeline/scripts/check_db.js
import pkg from 'pg';
import process from 'node:process';
const { Client } = pkg;

const connectionString = process.env.DATABASE_URL;

if (!connectionString) {
    throw new Error('Missing DATABASE_URL.');
}

async function main() {
    const client = new Client({ connectionString });
    try {
        await client.connect();
        const res = await client.query('SELECT id, title, filename FROM papers');
        console.log('Papers in DB:');
        console.table(res.rows);
    } catch (err) {
        console.error('Failed:', err);
    } finally {
        await client.end();
    }
}

main();
